### DISCUSSION TITLE
How to show Spatie Laravel Tags in table?

### DISCUSSION THREAD

**Discussion by bartclaeys** [filamentphp/filament#5333] [Help]
I successfully added Spatie Laravel Tags package and the Filament plugin. I can now add tags to my form, but how do I display these tags in my table? Tried the following, but it shows blank:

```
return $table
    ->columns([
        Tables\Columns\TagsColumn::make('tags')
]);
```

And added this to my model:

```
protected $casts = [
        'tags' => 'array',
    ];
```

**Comment by danharrin** [comment]
SpatieTagsColumn

**Comment by bartclaeys** [comment]
Unfortunately can't get it to work:

What I've done:

1. Added Spatie Laravel Tags package.
2. Added Filament Spatie Tags plugin.
3. Added `use HasTags;` to my model.
4. Added `protected $casts = ['tags' => 'array'];` to my model.
5. Added `SpatieTagsInput::make('tags')` to `$form->schema([]);` to my Filament resource.
6. Added `use Filament\Tables\Columns\SpatieTagsColumn;` at top of Filament resource.

This results in the error:

`Call to a member function filter() on null`

When I comment out `SpatieTagsInput::make('tags')` the error is gone.

Not sure what I am missing...

What's unclear to me is whether I need a `tags` field in my model's table. I don't think so (it would be duplicative because tags are save in their separate table).

**Reply by danharrin** [reply]
SpatieTagsColumn needs to be in the table, not SpatieTagsInput

**Reply by danharrin** [reply]
https://filamentphp.com/docs/2.x/spatie-laravel-tags-plugin/table-columns

**Reply by bartclaeys** [reply]
That's a typo on my side in above post, but my code does have SpatieTagsColumn(). It still doesn't work...

**Reply by danharrin** [reply]
Please share the Flare error URL

**Reply by bartclaeys** [reply]
Sorry, not using Flare, this is my first Laravel application...

**Reply by bartclaeys** [reply]
Fixed, I needed to define a type:

`SpatieTagsColumn::make('tags')->type('books');`

**Reply by danharrin** [reply]
> Sorry, not using Flare, this is my first Laravel application...

it comes pre-installed in Laravel, you dont need to sign up or anything

**Comment by eduardonwa** [comment]
i got the same error as op commented above, even went on to define the type but still getting "Call to a member function filter() on null"

**Reply by danharrin** [reply]
Please share the Flare error URL

**Reply by eduardonwa** [reply]
Here you go: https://flareapp.io/share/xmNljwx7

**Reply by danharrin** [reply]
Your custom Tag model implements HasTags? 🤔

**Reply by eduardonwa** [reply]
Yes it does

**Reply by danharrin** [reply]
But why? 🤔 Why would a tag have tags itself?

**Reply by eduardonwa** [reply]
I was just wondering if that would clear the issue, i could however remove the type if that's what you're referring to but i still get the same error. So basically this below throws the same filter method not being found 

```            ->columns([
                SpatieTagsColumn::make('tags'),
            ])
```

**Reply by danharrin** [reply]
I don't understand the error, it says that the tags property is null but isnt it a relationship?

**Reply by eduardonwa** [reply]
the filter method is null that's what it says

**Reply by danharrin** [reply]
No, the filter method is being *called on null*, the tags for the current model are null.
